grid-client/calculator/calculate.go

```diff
@@ -143,6 +152,294 @@ func calculateCU(cru, mru int64) float64 {
+}
+
+// unitToTFT converts unit-TFT (big.Int) to TFT
+func unitToTFT(units *big.Int) float64 {
+	result := new(big.Float).SetInt(units)
+	val, _ := result.Quo(result, big.NewFloat(UnitFactor)).Float64()
+	return val
+}
+
+// GetUnbilledAmountInTFT returns the amount unbilled for a given contract in TFT
+func (c *Calculator) getUnbilledAmountInTFT(contractID uint64) (float64, error) {
+	billingInfo, err := c.substrateConn.GetContractBillingInfo(contractID)
+	if err != nil && !errors.Is(err, substrate.ErrNotFound) {
+		return 0, err
+	}
+	var unbilled float64 = 0
+	if billingInfo.AmountUnbilled != types.U64(0) {
+		unbilled = float64(billingInfo.AmountUnbilled)
+	}
+	// amount unbilled is in unit-USD
+	unbilledUSD := unitToUSD(unbilled)
+
+	return c.USDtoTFT(unbilledUSD)
+}
+
+// Calculates the total overdue of contracts on a node in TFT
+func (c *Calculator) calculateTotalContractsOverdueOnNode(nodeID uint32, allowance time.Duration) (int64, error) {
+	contracts, err := c.substrateConn.GetNodeContracts(nodeID)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get contracts for node ID %d", nodeID)
+	}
+	var totalCost int64 = 0
+	for _, contract := range contracts {
+		cost, err := c.CalculateContractOverdue(uint64(contract), allowance)
+		if err != nil && err != ErrContractDeleted {
+			return 0, err
+		}
+		if err == ErrContractDeleted {
+			continue
+		}
+		totalCost += cost
+	}
+	return totalCost, nil
+}
+
+// Calculates the cost with a period in TFT.
+//
+// The period is the time since last updated in seconds with the provided allowance time.
+func (c *Calculator) calculatePeriodCostTFT(lastUpdatedSeconds time.Time, contract *substrate.Contract, allowance time.Duration) (float64, error) {
+	// Calculate the elapsed seconds since last billing
+	elapsedSeconds := math.Ceil(time.Since(lastUpdatedSeconds).Seconds())
+	totalPeriodSeconds := elapsedSeconds + allowance.Seconds()
+
+	contractMonthlyCostUSD, err := c.calculateContractCost(contract)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to calculate contract cost")
+	}
+
+	contractMonthlyCostTFT, err := c.USDtoTFT(contractMonthlyCostUSD)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to convert contract cost to TFT")
+	}
+	secondsPerMonth := 30 * 24 * 60 * 60 // 30 days * 24 hours * 60 minutes * 60 seconds
+	contractCostPerSecond := contractMonthlyCostTFT / float64(secondsPerMonth)
+	return contractCostPerSecond * totalPeriodSeconds, nil
+
+}
+
+// Calculates the cost of a contract per month in USD.
+func (c *Calculator) calculateContractCost(contract *substrate.Contract) (float64, error) {
+	if contract.ContractType.IsNameContract {
+		return c.calculateUniqueNameCost()
+	}
+
+	nodeID, err := getNodeID(contract)
+	if err != nil {
+		return 0, err
+	}
+
+	node, err := c.substrateConn.GetNode(nodeID)
+	if err != nil {
+		return 0, err
+	}
+
+	var nodeRentContract uint64
+
+	nodeRentContract, err = c.substrateConn.GetNodeRentContract(nodeID)
```

Comment:
you need this check only if the contract is node contract no? so we can move it to inside the if statement in line 320